pymeasure/instruments/hp/hp663X2A.py

```diff
@@ -0,0 +1,233 @@
+        "CURR?", "CURR %s",
+        """Control Output current limit setpoint""",
+    )
+
+    current_trigger_level = Instrument.control(
+        "CURR:TRIG?", "CURR:TRIG %s",
+        """Control the pending triggered current limit""",
+    )
+
+    current_protection_enabled = Instrument.control(
+        "CURR:PROT:STAT?", "CURR:PROT:STAT %s",
+        """Control current protecton.
+        Disables output if overcurrent is tripped. Off on reset""",
+        validator=strict_discrete_set,
+        values=BOOL_MAPPINGS,
+        map_values=True,
+    )
+
+    # #Measurement commands
+    current = Instrument.measurement(
+        "MEAS:CURR?",
+        """Measure the DC output current.""",
+    )
+
+    current_rms = Instrument.measurement(
+        "MEAS:CURR:ACDC?",
+        """Measure the AC+DC RMS output current.""",
+    )
+
+    current_min = Instrument.measurement(
+        "MEAS:CURR:MIN?",
+        """Measure the minimum DC output current from a sample.""",
+    )
+
+    current_max = Instrument.measurement(
+        "MEAS:CURR:MAX?",
+        """Measure the maximum DC output current from a sample.""",
+    )
+
+    voltage = Instrument.measurement(
+        "MEAS:VOLT?",
+        """Measure the dc output voltage.""",
+    )
+
+    voltage_rms = Instrument.measurement(
+        "MEAS:VOLT:ACDC?",
+        """Measure the ac+dc RMs output voltage.""",
+    )
+
+    voltage_min = Instrument.measurement(
+        "MEAS:VOLT:MIN?",
+        """Measure the low voltage level from a sample.""",
+    )
+
+    voltage_max = Instrument.measurement(
+        "MEAS:VOLT:MAX?",
+        """Measure the high voltage level from a sample.""",
+    )
+
+    # Output Trigger related commands
+    def init_single_output_trigger(self):
+        """Set the state of the triggering system to "wait-for-trigger".
+
+        The source will change the output levels to voltage_trigger_level and current_trigger_level.
+        It will return to Idle after a single trigger"""
+        self.write("INIT:NAME TRAN"),
+
+    def enable_cont_output_trigger(self):
+        """Set the state of the triggering system to "wait-for-trigger".
+
+        The source will change the output levels to voltage_trigger_level and current_trigger_level.
+        It will return to Idle after a single trigger"""
+        self.write("INIT:CONT:NAME TRAN, ON"),
+
+    trigger_output = Instrument.setting(
+        "*TRG",
+        """Set the trigger once""",
+    )
+
+    # Display related commands
+    display_enabled = Instrument.control(
+        "DISP?", "DISP %s",
+        """Control the display state.""",
+        validator=strict_discrete_set,
+        values=BOOL_MAPPINGS,
+        map_values=True,
+    )
+
+    display_mode = Instrument.control(
+        "DISP:MODE?", "DISP:MODE \"%s\"",
+        """Control the display, switch between normal and text display modes""",
+        validator=strict_discrete_set,
+        values=["NORM", "TEXT"],
+    )
```

Comment:
Strings are always difficult.
As you have just two options, you could use the following: name the property `display_text_mode_enabled`, which switches between text and normal.
With `values={True: "TEXT", False: "NORM"}` and `map_values=True`, it will do the translation in the background.